src/components/Atoms/Input/index.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { useRef, useEffect } from "react";
+import styled from "styled-components";
+import COLOR from "../../../variables/color.js";
+import FONTFAMILY from "../../../variables/font_family.js";
+import TEXT from "../../../variables/texts";
+
+export const Input = ({ onEditComplete, defaultValue }) => {
+  const inputRef = useRef(null);
+
+  const onkeypress = (e) => {
+    if (e.key === "Enter") {
+      inputRef.current.onblur();
+      inputRef.current.onkeypress("blur", () => {
+        onEditComplete(inputRef.current.value);
+        onkeypress(inputRef.current.value);
+      });
+    }
+  };
+
+  useEffect((props) => {
+    inputRef.current.value = defaultValue;
+    inputRef.current.focus();
+    inputRef.current.onblur = (e) => {
+      const inputText = e.target.value;
+      props.onEditComplete(inputText);
+    };
+  }, []);
+
+  return <StyledInput ref={inputRef} name="input" type="text" />;
```

Comment:
```suggestion
  return <StyledInput ref={inputRef} type="text" />;
```
`name` 属性はHTMLフォームにおいてサーバーに送信される情報のキーとして使われる値ですが、今回は入力された値のハンドリングはクライアント側だけで完結します。
そのため、`name` 属性はなくても大丈夫です。